packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/layout/layout.tsx

```diff
@@ -181,17 +420,18 @@ const layoutNodes = (allVisibleNodes: Node[], nodesToLayout: Node<NodeData>[]) =
         );
 
         // Update children position to be under the label and at the right padding.
-        directChildren.forEach((child, index) => {
-          child.position = {
-            x: rectangularNodePadding,
-            y: rectangularNodePadding + (labelElement?.getBoundingClientRect().height ?? 0) + rectangularNodePadding,
-          };
+        const directNodesChildren = directChildren.filter((child) => !child.data.isBorderNode);
+        directNodesChildren.forEach((child, index) => {
+          child.position = getChildNodePosition(allVisibleNodes, child, labelElement);
           if (index > 0) {
-            const previousSibling = directChildren[index - 1];
-            child.position = { ...child.position, x: previousSibling.position.x + (previousSibling.width ?? 0) + gap };
+            const previousSibling = directNodesChildren[index - 1];
+            child.position = getChildNodePosition(allVisibleNodes, child, labelElement, previousSibling);
           }
         });
 
+        // Set border node initial position
+        setBorderNodesInitialPosition(directChildren, labelElement);
```

Comment:
Instead of setting the border node position, you could:
1. Compute the footprint width of NORTH border nodes, then SOUTH border nodes, and do the same with the height of EAST and WEST border nodes.
2. Compare these fourth values with the label and children's footprint (should be renamed into contentBox size) to get the size of the current node.
3. Position border nodes

Reply:
I developed a function to calcul the contentBox that gives us the space of a node + the border nodes. It's used in the initial function : getChildrenFootprint
`
const getChildrenFootprint = (allVisibleNodes: Node<NodeData>[], children: Node[]): Rect => {
  const footPrint: Box = children.reduce(
    (currentFootPrint, node) => {
      const bounds = getNodeFootprint(allVisibleNodes, node);
      const nodeBox = rectToBox({
        x: bounds.x,
        y: bounds.y,
        width: bounds.width ?? 0,
        height: bounds.height ?? 0,
      });

      return getBoundsOfBoxes(currentFootPrint, nodeBox);
    },
    { x: Infinity, y: Infinity, x2: -Infinity, y2: -Infinity }
  );
  return boxToRect(footPrint);
};
`